db/migrations/20190726141224_create_flap_table_and_triggers.sql

```diff
@@ -134,32 +174,31 @@ VALUES (new_diff.bid_id,
         flap_bid_bid_before_block(new_diff.bid_id, new_diff.address_id, new_diff.header_id),
         (SELECT api.epoch_to_datetime(block_timestamp) FROM diff_block),
         flap_bid_time_created(new_diff.address_id, new_diff.bid_id))
-ON CONFLICT (block_number, bid_id, address_id) DO UPDATE SET guy = new_diff.guy;
+ON CONFLICT (block_number, bid_id, address_id) DO UPDATE SET guy = new_diff.guy
 $$
     LANGUAGE sql;
--- +goose StatementEnd
 
 -- +goose StatementBegin
-CREATE OR REPLACE FUNCTION maker.update_flap_guys_until_next_diff(new_diff maker.flap_bid_guy) RETURNS VOID
+CREATE OR REPLACE FUNCTION maker.update_flap_guys_until_next_diff(start_at_diff maker.flap_bid_guy, new_guy TEXT) RETURNS VOID
```

Comment:
What's the rationale for breaking out the `new_guy` param rather than just reading it off of the diff? Wondering if that could make it trickier to change the type from text -> int referencing addresses (id)

Reply:
With this new set of triggers, we had to make these `update_until_next_diff` functions more flexible. The `DELETE` triggers use the same `UPDATE` logic, but updates snapshots to the value of the next-most-recent diff, rather than using the value already contained within the diff being deleted. See line 222.

As a side-note, I've been deliberately avoiding including FKs like `address_id` in the trigger-tables where possible, in order to minimize the number of `JOIN`s necessary at request-time, since the trigger-tables were originally intended as performance-optimizations. Definitely open to reconsidering that approach though, especially where performance isn't likely to be as big of an issue.

Reply:
👍 to avoiding foreign keys in the trigger tables - would def prefer any necessary joins be handled in the trigger. Enabling re-use of the update logic in the `DELETE` triggers makes sense to me 👌